libpromises/eval_context.c

```diff
@@ -4047,3 +4054,32 @@ void EvalContextProfilingEnd(EvalContext *ctx, const Policy *policy)
 
     JsonDestroy(profiling);
 }
+
+// ##############################################################
+
+void EvalContextSetCommonEvalOrder(EvalContext *ctx, EvalContextEvalOrder eval_order)
+{
+    assert(ctx != NULL);
+    ctx->common = eval_order;
+}
+
+void EvalContextSetAgentEvalOrder(EvalContext *ctx, EvalContextEvalOrder eval_order)
+{
+    assert(ctx != NULL);
+    ctx->agent = eval_order;
+}
+
+bool EvalContextIsNormalOrder(EvalContext *ctx)
+{
+    assert(ctx != NULL);
+
+    bool agent_is_agent = (ctx->config->agent_type == AGENT_TYPE_AGENT);
+    bool common_is_normal = (ctx->common != EVAL_ORDER_TOP_DOWN);
+    bool agent_is_normal = (ctx->agent != EVAL_ORDER_TOP_DOWN);
+
+    if (ctx->agent != EVAL_ORDER_UNDEFINED)
+    {
+        return (agent_is_agent) ? agent_is_normal : common_is_normal;
+    }
+    return common_is_normal;
+}
```

Comment:
In this case, I think the booleans make this harder to read and the function could be written in a more clear way;

```suggestion
bool EvalContextIsNormalOrder(EvalContext *ctx)
{
    assert(ctx != NULL);
    
    if (ctx->config->agent_type != AGENT_TYPE_AGENT)
    {
        // Not cf-agent, so we ignore body agent control
        return (ctx->common_eval_order != EVAL_ORDER_TOP_DOWN);
    }
    
    // cf-agent
    if (ctx->agent_eval_order != EVAL_ORDER_UNDEFINED)
    {
        // evaluation_order from body agent control has priority, if defined
        return (ctx->agent_eval_order == EVAL_ORDER_CLASSIC);
    }
    
    // The fallback is to use what is defined in body common control,
    // or if not defined there either, default to true (normal order)
    return (ctx->common_eval_order != EVAL_ORDER_TOP_DOWN)
}
```